particles/PipeApps/RenderNotification.arcs

```diff
@@ -8,3 +8,11 @@ recipe NotificationTest
   create as output
   RenderNotification
     output = output
+
+particle RenderText in './source/RenderText.java'
+  consume root
+  modality log
+  modality alert
+
```

Comment:
I'm not sure I understand this, does this mean the output is broadcast to both modality renderers? I would have thought there would be a 1:1 correspondence between slot and modality, because the output you render is modality specific, e.g. DOM vs some other markup. Am I misunderstanding? When getTemplate() is called, how would the particle author direct the output to one renderer vs another?

Reply:
The way it is implemented today, particle may consume multiple slots, and particle may declare support of multiple modalities, however particle-spec doesn't express the relation between a slot and a modality. We've discussed this with the Runtime team and agreed that this how it should be done, but never had the time to tweak the implementation.

In theory, I don't think this is impossible that multiple modalities share the same format of the rendering content (like in this example, when the content is just text; or iirc slices support simplified html markup?)
This whole thing is a bit hypothetical, like the overall modality handling in this PR. For the sake of the demo, I decided to have multiple renderers handle this content. The particle author could limit the particle modality to a single modality (and write different particles for each modality).
But we can change this heuristics later, when we have specific particles and requirements.

Reply:
I don't mind landing it as is, but we do need to think about the multi-modal aspect more clearly.

Like you could have a recipe that renders to say, an Android Notification AND renders to say, a 3D surface. The markup for these is likely to be radically different, we can't really rely on modalities having the same format. The existing modality demos already diverge on this.

Reply:
I agree, we need to define this more clearly.
(not to argue with the above, but I think if the particle developer writes in a spec that their particle outputs rendering content in a specific format, we can could count on them using the right markup or dealing with the consequences)